docs/guide/javascript_and_css.md

```diff
@@ -8,6 +8,199 @@ parent: How-to guide
 
 While ViewComponent doesn't provide any built-in tooling to do so, it’s possible to include JavaScript and CSS alongside components.
 
+## Propshaft / Stimulus
+
+To use a [transpiler-less and bundler-less approach to JavaScript](https://world.hey.com/dhh/modern-web-apps-without-javascript-bundling-or-transpiling-a20f2755) (the default for Rails 8), Stimulus and CSS can be used inside ViewComponents one of two ways:
+
+### Upgrading a pre-Rails 8 app
+
+```ruby
+# Gemfile (then run `bundle install`)
+gem "importmap-rails" # JavaScript version/digests without transpiling/bundling
+gem "propshaft" # Load static assets like JavaScript/CSS/images without transpilation/webpacker
+gem "stimulus-rails" # Hotwire JavaScript approach
+```
+
+```js
+// app/javascript/controllers/application.js
+import { Application } from "@hotwired/stimulus"
+
+const application = Application.start()
+
+application.debug = false
+window.Stimulus   = application
+
+export { application }
+```
+
+```js
+// app/javascript/controllers/index.js
+import { application } from "controllers/application"
+```
+
+```ruby
+# config/importmap.rb
+pin "@hotwired/turbo-rails", to: "turbo.min.js"
+pin "application", preload: true
+pin "@hotwired/stimulus", to: "stimulus.min.js"
+pin "@hotwired/stimulus-loading", to: "stimulus-loading.js"
+```
+
+### Approach 1 - Default _app/components_ ViewComponent directory using named Stimulus controllers, no autoloading
+
+Locate CSS and Stimulus js with a ViewComponent. This example demonstrates a _HelloWorldComponent_ in an _examples_ namespace with a sidecar file naming approach:
+
+```console
+app/components
+├── ...
+├── examples
+|   ├── hello_world_component
+|   |   ├── hello_world_component_controller.js
+|   |   ├── hello_world_component.css
+|   |   └── hello_world_component.html.erb
+|   └── hello_world_component.rb
+├── ...
+```
+
+#### 1. Prepare _app/components_ as an asset path for css and ensure hot reloads of Stimulus JavaScript
+
+```ruby
+# config/application.rb
+config.assets.paths << "app/components"
```

Comment:
one small issue I has with this config in the past is it digests everything in this path even your erb files, one way to solve this is to get all the files in components and remove any mot asset files

Reply:
I'd be interested in a better approach. I'm not sure how to avoid this?

Reply:
Could we do something like this?

```ruby
config.assets.paths += Dir.glob("app/components/**/*.js")
```

Reply:
> I'd love to avoid autoloading all assets in that app/components directory but when I have tried various approaches like the one you have suggested, I've been unable to make that work.

What problems did you run into?

Reply:
`config.assets.paths` for the purpose of loading javascript with importmap-rails [expects this to contain an array of directories](https://github.com/rails/importmap-rails/blob/f588506eebc7b276f8b3539301633f93d2bda55d/lib/importmap/map.rb#L129-L141), not files.

Theoretically, you could probably create a directory structure for your view components like:
```console
app/components
├── ...
├── examples
|   ├── hello_world_component
|   |   ├── hello_world_component_javascript
|   |    |        ├── hello_world_component_controller.js
|   |   ├── hello_world_component.css
|   |   └── hello_world_component.html.erb
|   └── hello_world_component.rb
├── ...
```

and then do something like:
`config.assets.paths += Dir["app/components/**/"].select { |d| d.include?('javascript') }`

but that doesn't seem particularly nice.

Reply:
Hey @camertron, I note the thumbs up here but I'm not sure how to interpret that in order to update the pull request. Are you suggesting we look to try and demonstrate a more complicated folder structure or are you suggesting you're now comfortable with the perhaps unavoidable usage of:
`config.assets.paths << "app/components"`